### ISSUE TITLE
[Internal] Automate API Review tool to protect against breaking API changes

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-authentication-library-for-dotnet#1875]
ApiCompat  is used for Wilson and SAL, and we should extend to MSAL.

[Here ](https://identitydivision.visualstudio.com/IDDP/_taskgroup/5d4aa3fb-1633-42bb-9da3-ee1fcc3f8b9a) is the task group that we are using for cloud builds. If you are using only NuGet packages to get contract assemblies, you can disregard parameters prefixed with “ado”.

**Comment by gladjohn** [comment]
@bgavrilMS checking to see if we still need this task?

**Comment by bgavrilMS** [comment]
Yeah, I think it would be nice to explore this tool again. Most likely we want to wait until we get rid of xamarin - these kind of tools don't work too well with ancient targets. 

https://stackoverflow.com/questions/73257002/generate-text-file-of-public-api-of-net-library-for-versioning-and-compatibilit

**Comment by gladjohn** [comment]
Ok will revisit this after we remove xamarin support

**Comment by AngelinaAcero111** [comment]
we are lazy here

**Comment by keegan-caruso** [comment]
Consider using this: https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/baseline-version-validator

**Comment by bgavrilMS** [comment]
This is great, thanks for pointing this out @keegan-caruso. We should add this to MSAL, at least .net and .net fwk versions.

**Comment by pmaytak** [comment]
Another possible analyzer tool: https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md

**Comment by bgavrilMS** [comment]
Has been done.
